### PR TITLE
WIP: DOC: examples for docstrings in scipy.stats

### PR DESCRIPTION
**Pull request by brandondavid**
Work in progress.

This PR will cover any remaining items in gh-7168 for scipy.stats:
- [ ] anderson
- [ ] ansari
- [ ] bartlett
- [ ] combine_pvalues
- [ ] epps_singleton_2samp
- [ ] fligner
- [ ] friedmanchisquare
- [ ] kstatvar
- [ ] levene
- [ ] ranksums
- [x] trim1

In addition, some existing examples have been expanded:
- trimboth

### CODE REVIEW

**Comment by chrisb83** [comment]
Thanks, adding examples is great. Let me know once this PR is ready to be reviewed.

**Comment by brandondavid** [comment]
> Thanks, adding examples is great. Let me know once this PR is ready to be reviewed.

Thank you, @chrisb83, I'll let you know.  But also, please feel free to weigh in early if you see me heading in a bad direction.  I'll add some commentary up top to explain my assumptions and thinking as I go along.

File: scipy/stats/stats.py
Comment (mdhaber) [inline scipy/stats/stats.py:3107]: I think the example above demonstrates this already. Consider referring to it rather than including a separate example.

File: scipy/stats/stats.py
Comment (mdhaber) [inline scipy/stats/stats.py:3197]: Can this be added to `trimboth`? Maybe also add an example of right trimming.

**Comment by mdhaber** [comment]
@brandondavid thanks for starting this! Would you be interested in continuing? If you are able to work on it before January, I can review relatively quickly.

**Comment by mdhaber** [comment]
I merged main and implemented my comments. I'll merge if CI comes back green.

Found a bug in `trim1`'s axis behavior stemming from gh-4910. After this merges, I'll fix that using the `slice` stuff from `trimboth` and add the multidimensional array test from `trimboth`.

File: scipy/stats/_stats_py.py
Comment (mdhaber) [inline scipy/stats/_stats_py.py:3359]:
```suggestion

    >>> c = [2, 4, 6, 8, 0, 1, 3, 5, 7, 9]
```

File: scipy/stats/_stats_py.py
Comment (mdhaber) [inline scipy/stats/_stats_py.py:3432]:
```suggestion

    return an array of 8 values:
```

File: scipy/stats/_stats_py.py
Comment (mdhaber) [inline scipy/stats/_stats_py.py:3438]:
```suggestion

    array:
```

**Review by mdhaber — APPROVED** [review]
This should fix the rendering issues. I'll go ahead and merge.

Thanks @brandondavid! If you'd like to continue with your list, please do, but let's get this one in now.

File: scipy/stats/_stats_py.py
Comment (mdhaber) [inline scipy/stats/_stats_py.py:3441]:
```suggestion
    array:

```

**Review by mdhaber — COMMENTED** [review]
Oops

File: scipy/stats/_stats_py.py
Comment (mdhaber) [inline scipy/stats/_stats_py.py:3434]:
```suggestion
    return an array of 8 values:

```

**Pull request merged**
